### PR TITLE
New TypeDescriptor [20054]

### PR DESCRIPTION
**Pull request by richiware**
## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- *N/A* Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- [x] Any new/modified methods have been properly documented using Doxygen. 
- *N/A* Changes are ABI compatible. 
- *N/A* Changes are API compatible. 
- *N/A* New feature has been added to the `versions.md` file (if applicable).
- *N/A* New feature has been documented/Current behavior is correctly described in the documentation. 
    
- *N/A* Applicable backports have been included in the description.

## Reviewer Checklist
- [N/A] The PR has a milestone assigned.
- [x] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test this

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]

File: include/fastdds/dds/xtypes/dynamic_types/detail/dynamic_language_binding.hpp [hunk 1]
```diff
@@ -105,7 +105,7 @@ typedef std::map<ObjectName, ObjectName> Parameters;
  * @brief This class represents the enumeration ExtensibilityKind defined by the user in the IDL file.
  * @ingroup dynamic_language_binding
  */
-enum ExtensibilityKind : uint32_t
+enum class ExtensibilityKind : uint32_t
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/xtypes/dynamic_types/detail/dynamic_language_binding.hpp:108]: In progress

File: include/fastdds/dds/xtypes/dynamic_types/type_traits.hpp
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/xtypes/dynamic_types/type_traits.hpp:1]: It's useful to change implementation, for example, in tests

File: include/fastdds/dds/xtypes/dynamic_types/detail/type_traits.hpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef _FASTDDS_DDS_XTYPES_DYNAMIC_TYPES_DETAIL_TYPE_TRAITS_HPP_
+#define _FASTDDS_DDS_XTYPES_DYNAMIC_TYPES_DETAIL_TYPE_TRAITS_HPP_
+
+#include <memory>
+
+template<typename T>
+struct type_traits
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/dds/xtypes/dynamic_types/detail/type_traits.hpp]: Done in 5626133

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM. CI is not required (only headers that are not yet built into the project)

**Pull request merged**